resources/datasets/hmda/generate_msamd_csv.sh

```diff
@@ -0,0 +1,192 @@
+# and names. These fields will be pulled out and placed into a CSV formatted
+# file appropriate for appending to code_sheets/msamd.csv.
+#
+# The expected input format:
+#     code column: MSA code (e.g. 47894)
+#     name_column: MSA name (e.g. "WASHINGTON-ARLINGTON-ALEXANDRIA, DC-VA-MD-WV")
+#
+# The output format:
+#     year_column: (optional)
+#     code column: MSA code (e.g. 47894)
+#     name_column: pretty MSA name (e.g. "Washington, Arlington, Alexandria - DC, VA, MD, WV")
+#
+# Requirements:
+#     xlsx2csv (pip install xlsx2csv) required if using an xlsx file
+#     sed (Mac users: install a newer version of sed: `brew install gnu-sed`)
+#
+
+CSV_APPEND_FILE=
+CODE_COLUMN=1
+NAME_COLUMN=2
+INPUT_TYPE=csv
+PREPEND_YEAR=
+
+OPTIONS_GUIDE="Usage: convert-msalsx-to-html.sh [-c code-column] [-n name-column] [-a csv-append-file] [-t input-type] [-y year] input-file
+    -c	Default: $CODE_COLUMN
+    -n	Default: $NAME_COLUMN
+    -a	Append results to provided filename
+    -t  Options: csv, xlsx	Default: $INPUT_TYPE
+    -y  Prepend year as first column of csv
+"
+
+# Pull out the command line arguments (if any)
+while getopts ":c:n:a:t:y:h" opt; do
+    case $opt in
+        c)
+            CODE_COLUMN=$OPTARG
+            ;;
+        n)
+            NAME_COLUMN=$OPTARG
+            ;;
+        a)
+            CSV_APPEND_FILE=$OPTARG
+            ;;
+        t)
+            case "$OPTARG" in
+                csv|xlsx)
+                    INPUT_TYPE=$OPTARG
+                    ;;
+                *)
+                    echo "Invalid file type specified" >&2
+                    echo "$OPTIONS_GUIDE" >&2
+                    echo 1
+                    ;;
+            esac
+            ;;
+        y)
+            PREPEND_YEAR=$OPTARG
+            ;;
+        h)
+            echo "$OPTIONS_GUIDE" >&2
+            exit 0
+            ;;
+        \?)
+            echo "Invalid option: -$OPTARG" >&2
+            echo "$OPTIONS_GUIDE" >&2
+            exit 1
+            ;;
+    esac
+done
+
+shift $((OPTIND-1))  #This tells getopts to move on to the next argument.
+
+if [ $# -ne 1 ]; then
+     echo "Please include the input file"
+     echo "$OPTIONS_GUIDE" >&2
+     exit 1
+fi
+
+if [[ -n "$CSV_APPEND_FILE" && -z "$PREPEND_YEAR" ]] ; then
+    echo "If appending to an existing CSV file, please define the year"
+    exit 1
+fi
+
+echo $INPUT_TYPE
+if [ $INPUT_TYPE = 'xlsx' ]; then
+    # Use xlsx2csv to convert the xlsx file: https://github.com/dilshod/xlsx2csv
+    # If xlsx2csv is not installed, exit
+    command -v xlsx2csv 2> /dev/null || {
+        echo >&2 "'xlsx2csv' is required for this script to execute. To install, run:
+pip install xlsx2csv 
+Note that root access may be required to run pip commands. Aborting.";
+        exit 1;
+    }
+fi
+
+# NOTE - if using OSX, you'll need to install a newer version of sed to support these queries.
+# `brew install gnu-sed` will create the gsed executable
+sed_bin='sed'
```

Comment:
Good idea.  I couldn't find a good way to check sed version (mac sed doesn't have a '-v' option...) but maybe just saying "if you're a mac and you don't have gsed, warning!"